internal/controller/drclusterconfig_controller.go

```diff
@@ -272,12 +288,71 @@ func (r *DRClusterConfigReconciler) processCreateOrUpdate(
 		return ctrl.Result{Requeue: true}, err
 	}
 
+	if err := r.validateS3Profiles(ctx, log, drCConfig); err != nil {
+		log.Info("Reconcile error", "error", err)
+		setDRClusterConfigConfigurationProcessedCondition(&drCConfig.Status.Conditions, drCConfig.Generation,
```

Comment:
This should call `setDRClusterConfigConfigurationProcessedCondition` and not update the processed condition.

I see that the function `validateS3Profiles` is updating the specific condition, so here we should not update or change the processed condition.